File: CleverFerretV2/feature/reader/src/main/java/com/cleverferret/v2/feature/reader/usecase/ReaderAiIntegrationUseCase.java
package com.cleverferret.v2.feature.reader.usecase;

import com.cleverferret.v2.core.common.result.IntegrationResult;
import com.cleverferret.v2.feature.ai.api.ReaderAiUseCases;
import com.cleverferret.v2.feature.ai.model.AiGenerationResult;

public final class ReaderAiIntegrationUseCase {
    private final ReaderAiUseCases readerAiUseCases;

    public ReaderAiIntegrationUseCase(ReaderAiUseCases readerAiUseCases) {
        this.readerAiUseCases = readerAiUseCases;
    }

    public IntegrationResult<AiGenerationResult> summarizeCurrentChapter(String chapterText) {
        return readerAiUseCases.summarizeSelection("feature:reader", chapterText);
    }

    public IntegrationResult<AiGenerationResult> analyzeCurrentCharacters(String chapterText) {
        return readerAiUseCases.analyzeCharacters("feature:reader", chapterText);
    }

    public IntegrationResult<AiGenerationResult> buildChapterMindMap(String chapterText) {
        return readerAiUseCases.generateMindMap("feature:reader", chapterText);
    }
}

File: CleverFerretV2/feature/reader/src/main/java/com/cleverferret/v2/feature/reader/usecase/ReaderAiIntegrationUseCase.kt
package com.cleverferret.v2.feature.reader.usecase

import com.cleverferret.v2.core.common.result.IntegrationResult
import com.cleverferret.v2.feature.ai.api.ReaderAiUseCases
import com.cleverferret.v2.feature.ai.model.AiGenerationResult

class ReaderAiIntegrationUseCase(private val readerAiUseCases: ReaderAiUseCases) {
    fun summarizeCurrentChapter(chapterText: String): IntegrationResult<AiGenerationResult> =
        readerAiUseCases.summarizeSelection("feature:reader", chapterText)

    fun analyzeCurrentCharacters(chapterText: String): IntegrationResult<AiGenerationResult> =
        readerAiUseCases.analyzeCharacters("feature:reader", chapterText)

    fun buildChapterMindMap(chapterText: String): IntegrationResult<AiGenerationResult> =
        readerAiUseCases.generateMindMap("feature:reader", chapterText)
}
